cmd/titus-metadata-service/main.go

```diff
@@ -235,29 +202,44 @@ func main() {
 			EnvVar:      "X_FORWARDED_FOR_BLOCKING_MODE",
 			Destination: &xFordwardedForBlockingMode,
 		},
+		cli.StringFlag{
+			Name:        "peer-namespace",
```

Comment:
I am not a fan of adding this logic to the metadata proxy. I liked it better when we could have a separation of concerns, and the process which was wiring up the network namespaces was different than the one doing the IMDS work.

The reasoning behind this is a sense of "purity" of the IMDS code, in that it doesn't do any systems or OS-level stuff.

Reply:
This process is not accessible from the pod. If it is, we have a host compromise.